contracts/Core/RewardManager.sol

```diff
@@ -119,10 +123,15 @@ contract RewardManager is Initializable, Constants, RewardManagerParams, IReward
             uint32 medianLastEpoch = mediansLastEpoch[i];
             if (medianLastEpoch == 0) continue;
             uint64 prod = age * voteValueLastEpoch;
+            // slither-disable-next-line calls-loop
+            uint16 tolerance = assetManager.getCollectionTolerance(i);
+            tolerance = tolerance <= maxTolerance ? tolerance : maxTolerance;
+            uint64 maxVoteTolerance = medianLastEpoch + ((medianLastEpoch * tolerance) / BASE_DENOMINATOR);
+            uint64 minVoteTolerance = medianLastEpoch - ((medianLastEpoch * tolerance) / BASE_DENOMINATOR);
             // if (voteWeightLastEpoch > 0) {
-            if (voteValueLastEpoch > medianLastEpoch) {
+            if (voteValueLastEpoch > maxVoteTolerance) {
                 penalty = penalty + (prod / medianLastEpoch - age);
```

Comment:
Penalties must be smoother.

Reply:
Done https://github.com/razor-network/contracts/pull/628/commits/3de20dd7c241a498284f6d7053828cc1bf6f2fda